File: day_1/day_1.py
def part_1(a):
    n = 0
    for i, v in enumerate(a):
        if i < len(a) - 1:
            if v < a[i+1]:
                n += 1
    return n

def part_2(a):
    n = 0
    for i, v in enumerate(a):
        if i < len(a) - 3:
            if v + a[i+1] + a[i+2] < a[i+1] + a[i+2] + a[i+3]:
                n += 1
    return n


f = open("input.txt", "r")
a = list(map(lambda x: int(x), f.read().split()))
print(part_1(a))
print(part_2(a))

File: day_1/day_1.js
const fs = require('fs')

const part_1 = arr => {
	let n = 0;
	for (let i = 0; i < arr.length - 1; i++) {
		if (arr[i] < arr[i+1]) {
			n++
		}
	}
	return n;
}

const part_2 = arr => {
	let n = 0;
	for (let i = 0; i < arr.length - 3; i++) {
		if (arr[i] + arr[i+1] + arr[i+2] < arr[i+1] + arr[i+2] + arr[i+3]) {
			n++
		}
	}
	return n;
}

fs.readFile('input.txt', 'utf-8', (err, data) => {
	if (err) throw err;
	let arr = data.split('\n').filter(s => s != '').map(x => parseInt(x));
	console.log(part_1(arr));
	console.log(part_2(arr));
})
